--- configure.in	2014-06-04 13:48:26.759547322 -0400
+++ configure.in.new	2014-06-04 14:10:39.801067464 -0400
@@ -700,17 +697,18 @@
 AC_CHECK_FUNCS([daq_hup_apply] [daq_acquire_with_meta])
 
 AC_MSG_CHECKING([for daq address space ID])
-AC_RUN_IFELSE(
-[AC_LANG_PROGRAM(
-[[
-#include <daq.h>
-]],
-[[
-   DAQ_PktHdr_t hdr;
-   hdr.address_space_id = 0;
-]])],
-[have_daq_address_space_id="yes"],
-[have_daq_address_space_id="no"])
+AC_LINK_IFELSE(
+    [AC_LANG_PROGRAM(
+    [[
+        #include <daq.h>
+    ]],
+    [[
+           DAQ_PktHdr_t hdr;
+           hdr.address_space_id = 0;
+    ]])],
+    [have_daq_address_space_id="yes"],
+    [have_daq_address_space_id="no"]
+)
 AC_MSG_RESULT($have_daq_address_space_id)
 if test "x$have_daq_address_space_id" = "xyes"; then
     AC_DEFINE([HAVE_DAQ_ADDRESS_SPACE_ID],[1],
